src/server/services/UrlManagementService.ts

```diff
@@ -47,7 +47,10 @@ export class UrlManagementService implements UrlManagementServiceInterface {
 
     const existsShortUrl = await this.urlRepository.findByShortUrl(shortUrl)
     if (existsShortUrl) {
-      throw new AlreadyExistsError(`Short link "${shortUrl}" already exists`)
+      const owner = await this.userRepository.findUserByUrl(shortUrl)
+      throw new AlreadyExistsError(
+        `Short link "${shortUrl}" is owned by ${owner?.email}`,
+      )
```

Comment:
Looks like the message contained in the error is also the UI message displayed to the user, which is very implicit. Do you think a refactor should be performed at some point to make this explicit?

Also, I don't think you have to `findByShortUrl` anymore, since you know the link already exists if `findUserByUrl` returns an owner.

Reply:
I don't think the refactor is necessary? The error thrown is made into the appropriate response which is then read by the UI. I don't think you would want a specialised set of error response readers, one for each kind of error.